### PR TITLE
Use pip for vLLM tutorial

### PR DESCRIPTION
**Pull request by dyastremsky**
In the vLLM tutorial, we use Conda for setting up and loading the environment, which allows the vLLM model to use a different CUDA version then Triton server. In vLLM 0.1.7, that is no longer necessary for the current version of Triton.

As a result, we can speed up the environment set up and simplify the tutorial steps by using Pip to install vLLM inside the Triton container instead. This pull request switches to that approach.

As a small improvement, this pull request also reduces the default GPU memory usage to 50%, since the example model is small and does not need as much GPU memory. This will reduce out-of-memory errors resulting from users having other processes using the GPUs and show how to use this parameter.

### CODE REVIEW

File: Quick_Deploy/vLLM/README.md [hunk 1]
```diff
@@ -84,17 +55,18 @@ and
...
 Read through the documentation in [`model.py`](model_repository/vllm/1/model.py) to understand how
 to configure this sample for your use-case.
 
-Using the container you already started, run the following command:
+Run the following commands to build the image and start the server container:
 
 ```
-tritonserver --model-store=model_repository
+docker build -t triton_vllm .
+docker run --gpus all -it --rm -p 8000:8000 -p 8001:8001 -p 8002:8002 --shm-size=8G --ulimit memlock=-1 --ulimit stack=67108864 -v ${PWD}:/work -w /work nvcr.io/nvidia/tritonserver:23.08-py3 tritonserver --model-store ./model_repository
```
Comment (rmccorm4) [inline Quick_Deploy/vLLM/README.md:69]:
```suggestion
docker run --gpus all -it --rm -p 8000:8000 -p 8001:8001 -p 8002:8002 --shm-size=8G --ulimit memlock=-1 --ulimit stack=67108864 -v ${PWD}:/work -w /work triton_vllm tritonserver --model-store ./model_repository
```

File: Quick_Deploy/vLLM/README.md [hunk 2]
```diff
@@ -84,17 +55,18 @@ and
 [EngineArgs](https://github.com/vllm-project/vllm/blob/32b6816e556f69f1672085a6267e8516bcb8e622/vllm/engine/arg_utils.py#L11)
 for supported key-value pairs.
 
-*Note*: vLLM greedily consume upto 90% of the GPU's memory under default settings. You can provide
+*Note*: vLLM greedily consume up to 50% of the GPU's memory under default settings. You can provide
```
Comment (rmccorm4) [inline Quick_Deploy/vLLM/README.md:58]:
```suggestion
*Note*: vLLM greedily consumes up to 50% of the GPU's memory under default settings. You can tweak this behavior using
```

**Review by rmccorm4 — COMMENTED** [review]
LGTM other than minor comment above

File: Quick_Deploy/vLLM/README.md [hunk 3]
```diff
@@ -84,17 +55,18 @@ and
 [EngineArgs](https://github.com/vllm-project/vllm/blob/32b6816e556f69f1672085a6267e8516bcb8e622/vllm/engine/arg_utils.py#L11)
 for supported key-value pairs.
 
-*Note*: vLLM greedily consume upto 90% of the GPU's memory under default settings. You can provide
+*Note*: vLLM greedily consumes up to 50% of the GPU's memory under default settings. You can tweak this behavior using
```
Comment (tanmayv25) [inline Quick_Deploy/vLLM/README.md:58]: I think this should still be 90%. 
https://github.com/vllm-project/vllm/blob/main/vllm/engine/arg_utils.py#L27

**Comment by dyastremsky** [comment]
I validated this builds and runs. With the new pip approach, we are also able to reduce the shared memory size flag to 1GB due to not needing to unpack the Conda environment. I also removed the unused ports and the no-longer-used environment generation script.

File: Quick_Deploy/vLLM/README.md [hunk 4]
```diff
@@ -84,17 +55,18 @@ and
...
 Read through the documentation in [`model.py`](model_repository/vllm/1/model.py) to understand how
 to configure this sample for your use-case.
 
-Using the container you already started, run the following command:
+Run the following commands to build the image and start the server container:
 
 ```
-tritonserver --model-store=model_repository
+docker build -t triton_vllm .
+docker run --gpus all -it --rm -p 8001:8001 --shm-size=1G --ulimit memlock=-1 --ulimit stack=67108864 -v ${PWD}:/work -w /work triton_vllm tritonserver --model-store ./model_repository
```
Comment (rmccorm4) [inline Quick_Deploy/vLLM/README.md:69]: Lowering to `--shm-size=1G` requirement is nice, more in line with requirements for some other popular examples, such as TGI:
```
docker run --gpus all --shm-size 1g -p 8080:80 -v $volume:/data ghcr.io/huggingface/text-generation-inference:1.0.3 --model-id $model
```

File: Quick_Deploy/vLLM/README.md [hunk 5]
```diff
@@ -84,17 +55,18 @@ and
...
 
 Read through the documentation in [`model.py`](model_repository/vllm/1/model.py) to understand how
 to configure this sample for your use-case.
 
-Using the container you already started, run the following command:
+Run the following commands to build the image and start the server container:
 
 ```
-tritonserver --model-store=model_repository
+docker build -t triton_vllm .
```
Comment (tanmayv25) [inline Quick_Deploy/vLLM/README.md:68]: Can the part building docker container be made a separate step?
You can see the readme here for reference:
https://github.com/triton-inference-server/tutorials/blob/14aec329fb8a7916ef579a470f3fec7504f06611/Quick_Deploy/vLLM/README.md

**Review by rmccorm4 — APPROVED** [review]

File: Quick_Deploy/vLLM/README.md (hunk 3 above)
Comment (tanmayv25) [reply Quick_Deploy/vLLM/README.md:58]: Just add a line towards end like...

settings via `vllm_engine_args.json`. For the purpose of the demo model we are setting this to 0.50.

Comment (dyastremsky) [reply Quick_Deploy/vLLM/README.md]: Good catch! Updated the wording.

File: Quick_Deploy/vLLM/README.md (hunk 5 above)
Comment (dyastremsky) [reply Quick_Deploy/vLLM/README.md]: Sure, done.

File: Quick_Deploy/vLLM/README.md [hunk 6]
```diff
@@ -38,36 +38,18 @@ Triton Inference Server using Triton's [Python backend](https://github.com/trito
...
-Running vLLM within Triton container requires us to provide
-[custom execution environment](https://github.com/triton-inference-server/python_backend#creating-custom-execution-environments)
-with vLLM and other package dependencies. The provided script should build the package environment
-for you which will be used to load the model in Triton.
+We will build a new container image derived from tritonserver:23.08-py3 with vLLM.
 
 ```
-docker run --gpus all -it --rm -p 8000:8000 -p 8001:8001 -p 8002:8002 --shm-size=8G --ulimit memlock=-1 --ulimit stack=67108864 -v ${PWD}:/work -w /work nvcr.io/nvidia/tritonserver:23.08-py3 bash
-./gen_vllm_env.sh
+docker build -t triton_vllm .
```
Comment (tanmayv25) [inline Quick_Deploy/vLLM/README.md:46]: Let's rename this to `tritonserver`

File: Quick_Deploy/vLLM/README.md [hunk 7]
```diff
@@ -38,36 +38,18 @@ Triton Inference Server using Triton's [Python backend](https://github.com/trito
...
-`-- vllm
-    |-- 1
-    |   `-- model.py
-    |-- config.pbtxt
-    |-- triton_python_backend_stub
-    |-- vllm_engine_args.json
-    `-- vllm_env
-
-```
+## Step 2: Start Triton Inference Server
```
Comment (tanmayv25) [inline Quick_Deploy/vLLM/README.md:52]: I think it would be a good idea to display structure of model_repository anyways.

Comment (dyastremsky) [reply Quick_Deploy/vLLM/README.md:52]: Sounds good, added.

File: Quick_Deploy/vLLM/README.md (hunk 6 above)
Comment (dyastremsky) [reply Quick_Deploy/vLLM/README.md]: Sure thing, renamed.

**Review by tanmayv25 — APPROVED** [review]

**Comment by dyastremsky** [comment]
Thanks for approving, @tanmayv25! Would you also be able to merge this? I don't have permissions.

CC: @dzier, if it would be possible for me to get contributing/merging permissions for this repository, it would be very helpful. I am working on a few vLLM-related tickets.

File: Quick_Deploy/vLLM/model_repository/vllm/vllm_engine_args.json [hunk 8]
```diff
@@ -1,4 +1,5 @@
 {
     "model":"facebook/opt-125m",
-    "disable_log_requests": "true"
+    "disable_log_requests": "true",
+    "gpu_memory_utilization": 0.5
```
Comment (rmccorm4) [inline Quick_Deploy/vLLM/model_repository/vllm/vllm_engine_args.json:4]: Add this line to the README section that's showing json contents as well

Comment (dyastremsky) [reply Quick_Deploy/vLLM/model_repository/vllm/vllm_engine_args.json:4]: Good catch, thanks for spotting this. Added.

**Review by tanmayv25 — APPROVED** [review]

**Pull request merged**